networking_cisco/plugins/cisco/cfg_agent/service_helpers/routing_svc_helper_aci.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2014 Cisco Systems, Inc.  All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+from oslo_utils import excutils
+import pprint as pp
+
+from neutron.common import constants as l3_constants
+
+from networking_cisco.plugins.cisco.cfg_agent import cfg_exceptions
+from networking_cisco.plugins.cisco.cfg_agent.service_helpers import (
+    routing_svc_helper as helper)
+from networking_cisco.plugins.cisco.common import (cisco_constants as
+                                                   c_constants)
+from networking_cisco.plugins.cisco.extensions import routerrole
+
+ROUTER_ROLE_ATTR = routerrole.ROUTER_ROLE_ATTR
+LOG = logging.getLogger(__name__)
+
+
+class RoutingServiceHelperAci(helper.RoutingServiceHelper):
+
+    def __init__(self, host, conf, cfg_agent):
+        super(RoutingServiceHelperAci, self).__init__(
+            host, conf, cfg_agent)
+        self._router_ids_by_vrf = {}
+
+    def _process_router(self, ri):
```

Comment:
This API shares lots of codes with the parent's. One way to refactor it is to create a helper function in the parent class with 1 more parameter indicating if its for ACI for not. Then in that helper function, if ACI is set to true, then u do all the extra stuff./check u like to do there. Then the parent's _process_router() will just call that helper with ACI flag set to false. While the _process_router() here will call that helper with ACI flag set to true.